[EDU-3462] feat: Fauna template guide (#489)
aziontech/docs: src/content/docs/en/pages/guides/marketplace/templates/fauna-template/fauna-template.mdx

---
title: How to deploy a database on the edge with the Fauna template
description: This template helps you to deploy and manage a Fauna-integrated database on the edge.
meta_tags: templates, guides, Azion Marketplace
namespace: docs_guides_fauna_template
permalink: /documentation/products/guides/fauna-template/
---
import Badge from '~/components/Badge.astro';

<Badge variant="accent">
Preview
</Badge>

The **Fauna** template allows you to deploy a database directly on the **Azion Edge Platform**. The template includes the creation of a static interface to facilitate you to create, retrieve, update, and delete items in your collections.

---

## Requirements

Before using this template, you need to: 

- Create a [Fauna account](https://dashboard.fauna.com/register).
- Create a database in the [Fauna Dashboard](https://docs.fauna.com/fauna/current/get_started/dashboard) and populate it with `demo data`, selecting this option while creating.

> - Create a new [collection](https://docs.fauna.com/fauna/current/cookbook/data_model/collections) called `Posts` within your database to use it with this template. 

Reviewer: ```suggestion
- Create a new [collection](https://docs.fauna.com/fauna/current/cookbook/data_model/collections) named `Posts` within your database to use it with this template. 
```

- Generate a [Fauna secret key](https://docs.fauna.com/fauna/v4/security/keys?lang=shell) for your database.
- Generate your [GitHub personal token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-personal-access-token-classic).

:::note

> While generating your GitHub personal token, grant that your **scope** has the permission to authorize an OAuth app or a personal token to access to public and private repositories, including read and write access to code. You must also enable the **workflow** option to allow adding and updating GitHub Actions workflow files.

Reviewer 2: ```suggestion
While generating your GitHub personal token, grant that your **scope** has the permission to authorize an OAuth app or a personal token to access public and private repositories, including read and write access to code. You must also enable the **workflow** option to allow adding and updating GitHub Actions workflow files.
```

> Note that permissions can be scoped either to a user an organization or a repository. Read the [Creating a personal access token documentation](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-personal-access-token-classic) for more information.

Reviewer: ```suggestion
Note that permissions can be scoped either to a user, an organization, or a repository. Read the [Creating a personal access token documentation](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-personal-access-token-classic) for more information.
```

- An Azion domain to access your edge application. You can also set up a custom domain to run on this edge application.
- An edge function containing custom logic.
- A GitHub repository for your project. The repository includes a GitHub Action that, by being activated, guarantees a [continuous deployment workflow](#continuous-deployment). 
- A static interface, based on HTML, to manage the database.

Additionally, to guarantee the optimal performance of this template, it's necessary to activate:

- [Edge Functions](https://www.azion.com/en/documentation/products/edge-application/edge-functions/)
- [Application Acceleration](https://azion.com/en/documentation/products/edge-application/application-acceleration/)

You need to activate these modules separately via RTM. To do so:

1. [Access RTM](https://manager.azion.com/).
2. On the upper-left corner, select the three horizontal lines to open the **Products menu**.
3. In the **BUILD** section, select **Edge Application**.
4. You'll be redirected to the **Edge Application** page.
- It lists all the edge applications you've created. 
5. Find the edge application related to your template and select it. 

> 6. In the **Main Settings** tab, find the **Edge Application Modules** section and active the switch for the product you want to enable.

Author: ```suggestion
6. In the **Main Settings** tab, find the **Edge Application Modules** section and activate the switch for the module you want to enable.
```

:::caution[Warning]
If these modules are activated, the execution of this template could generate usage-related costs. Check the [pricing page](https://www.azion.com/en/pricing/) for more information. 

Any cost generated by Fauna usage will be processed and billed separately on the Fauna platform. Visit the [Fauna documentation](https://docs.fauna.com/fauna/current/plan_billing/) for more details.
:::

---

## Managing your database collection

You can use the Azion domain to access it through the browser, as explained on the [deployment section](#deploying-the-template), and interact with it through the static interface that was created.

> By using this interface, you're able to **create**, **retrieve**, **update**, and **delete** items in your `Posts` collection, which is now linked to this new application. To do so, the interface provides a input area and icons buttons to manage your data as required.

Reviewer 2: ```suggestion
By using this interface, you're able to **create**, **retrieve**, **update**, and **delete** items in your `Posts` collection, which is now linked to this new application. To do so, the interface provides an input area and icon buttons to manage your data as required.
```

All changes will be reflected on your application and your Fauna database as well. 

---

## Managing the template

Considering that this initial setup may not be optimal for your specific edge application, all settings can be customized any time you need by using RTM.

To manage and edit your edge application's settings, proceed as follows:

1. [Access RTM](https://manager.azion.com/).
2. On the upper-left corner, select the three horizontal lines to open the **Products menu**.
3. In the **BUILD** section, select **Edge Application**.
4. You'll be redirected to the **Edge Application** page.
- It lists all the edge applications you've created. 

> 5. Find the edge application related to the Hello World template and select it. 

Reviewer: ```suggestion
5. Find the edge application related to the **Hello World** template and select it. 
```
